nuget/Microsoft.Windows.CppWinRT.targets

```diff
@@ -395,8 +402,11 @@ $(XamlMetaDataProviderPch)
 
     <!--Insert Midl /references to Platform WinMDs, library reference WinMDs, and direct reference WinMDs-->
     <Target Name="CppWinRTSetMidlReferences"
-            DependsOnTargets="GetCppWinRTPlatformWinMDReferences;GetCppWinRTDirectWinMDReferences;GetCppWinRTProjectWinMDReferences;$(CppWinRTSetMidlReferencesDependsOn)">
-        <ItemGroup Condition="'$(CppWinRTModernIDL)' != 'false'">
+            Condition="'$(CppWinRTModernIDL)' != 'false'"
+            DependsOnTargets="GetCppWinRTPlatformWinMDReferences;GetCppWinRTDirectWinMDReferences;GetCppWinRTProjectWinMDReferences;$(CppWinRTSetMidlReferencesDependsOn)"
+            Inputs="@(CppWinRTDirectWinMDReferences);@(CppWinRTStaticProjectWinMDReferences);@(CppWinRTDynamicProjectWinMDReferences);@(CppWinRTPlatformWinMDReferences)"
```

Comment:
It's not directly related to this PR, but you might consider adding `$(MSBuildAllProjects)` to the `Inputs` in these targets. That's a common pattern used as a proxy for "changing a property or item". A common case where this is helpful is "deleting a reference"--the current project output is up to date with all of the inputs, but should be rebuilt anyway because the list of inputs has changed. There's no general solution to this in MSBuild but `$(MSBuildAllProjects)` helps.

Reply:
Cool! I will look into this.

Reply:
@Rainersigwald I am trying to find example of how the Midl or the C++/C# compiler targets handle deletes from the Compile or ClCompile itemgroup but it doesn't look like they look at `$(MSBuildAllProjects)`. Am I missing something?

Reply:
C# is here: https://github.com/dotnet/roslyn/blob/b0df544ac729ff09031d3cf38e2cbfb3d69436a5/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L13

Note that with C#/VB, we have another more sophisticated approach, where we write the hash of the list to a file and include that as an input (microsoft/msbuild#1328). That's often overkill, and still misses the "some property passed to the compiler changed" scenario, so you still need `$(MSBuildAllProjects)`.